File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minstock = prices[0];
        int maxprofit = 0;
        for(int i=1; i<prices.length; i++){
            minstock = Math.min(minstock, prices[i]);
            int profit = prices[i]-minstock;
            maxprofit = Math.max(maxprofit, profit);
        }
        return maxprofit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let minprice = Infinity;
    let maxprice = 0;

    for(let i = 0; i< prices.length; i++){
        if(prices[i] < minprice) {
            minprice = prices[i];
        }
        else{
        let profit = prices[i] - minprice;
        maxprice = Math.max(profit, maxprice);
        }
        
    }
    return maxprice;
};